warehouse/manage/views/__init__.py

```diff
@@ -544,27 +536,21 @@ def validate_totp_provision(self):
             )
             return HTTPSeeOther(self.request.route_path("manage.account"))
 
-        totp_secret = self.user_service.get_totp_secret(self.request.user.id)
-        if totp_secret:
-            self.request.session.flash(
-                "Account cannot be linked to more than one authentication "
-                "application at a time",
-                queue="error",
-            )
-            return HTTPSeeOther(self.request.route_path("manage.account"))
-
         form = ProvisionTOTPForm(
             MultiDict(**self.request.POST),
             totp_secret=self.request.session.get_totp_secret(),
         )
 
         if form.validate():
+            old_totp_secret = self.user_service.get_totp_secret(self.request.user.id)
             self.user_service.update_user(
                 self.request.user.id, totp_secret=self.request.session.get_totp_secret()
             )
             self.request.session.clear_totp_secret()
             self.request.user.record_event(
-                tag=EventTag.Account.TwoFactorMethodAdded,
+                tag=EventTag.Account.TwoFactorMethodEdited
+                if old_totp_secret
+                else EventTag.Account.TwoFactorMethodAdded,
                 request=self.request,
                 additional={"method": "totp"},
             )
```

Comment:
Instead of introducing a new event type, I think we should just emit two back-to-back events, "Removed" + "Added".